hydromt/data_adapter/dataframe.py

```diff
@@ -196,52 +187,20 @@ def get_data(
         based on the properties of this DataFrameAdapter. For a detailed
         description see: :py:func:`~hydromt.data_catalog.DataCatalog.get_dataframe`
         """
-        kwargs = self._parse_args()
-        df = self._load_data(variables, **kwargs)
-        df = DataFrameAdapter.slice_data(df, time_tuple)
-        df = self._uniformize_data(df)
-        return df
-
-    def _load_data(self, variables, **kwargs):
-        df = self._read_data(**kwargs)
-        df = self._rename_vars(df, variables)
-        return df
-
-    def _uniformize_data(self, df):
-        df = self._apply_unit_conversion(df)
-        df = self._set_meta_data(df)
-        return df
-
-    @staticmethod
-    def slice_data(df, time_tuple):
-        """Return a sliced DataFrame.
-
-        Parameters
-        ----------
-        df : pd.DataFrame
-            the dataframe to be sliced.
-        time_tuple : tuple of str, datetime, optional
-            Start and end date of period of interest. By default the entire time period
-            of the dataset is returned.
-
-        Returns
-        -------
-        pd.DataFrame
-            Tabular data
-        """
-        return DataFrameAdapter._slice_temporal_dimension(df, time_tuple)
-
-    @staticmethod
-    def _slice_temporal_dimension(df, time_tuple):
-        if time_tuple is not None and np.dtype(df.index).type == np.datetime64:
-            logger.debug(f"DataFrame: Slicing time dime {time_tuple}")
-            df = df[df.index.slice_indexer(*time_tuple)]
-            if df.size == 0:
-                raise IndexError("DataFrame: Time slice out of range.")
-
+        # load data
+        fns = self._resolve_paths(variables)
+        df = self._read_data(fns, logger=logger)
+        # rename variables and parse nodata
+        df = self._rename_vars(df)
+        df = self._set_nodata(df)
+        # slice data
+        df = self._slice_data(df, variables, time_tuple, logger=logger)
```

Comment:
I'm also not sure this should be a private method? it seems like something that python users might want to use, or am I wrong in that?

Reply:
> `_slice_data` is a static method so you should call it as `DataframeAdapter._slice_data` I'm rather surprised this works doesn't throw a syntax error.

I think it's ok to call the static method `_slice_data` with `self._slice_data` within the context of the the class. It will be based on the instance of that class, but since it does not use any properties of the class it returns the same as when calling `DataFrameAdapter._slice_data`. I only used `self` as I find it easier to read and it is shorter.  See also, this [SO question](https://stackoverflow.com/questions/65461959/calling-a-static-method-with-self-vs-class-name)

> I'm also not sure this should be a private method? it seems like something that python users might want to use, or am I wrong in that?

I do think it should be a private method, but we can discuss about it. In my opinion the public methods of the Adapters should only be the `get_data`, `to_file` and `to_dict` methods as the purpose is really to access data in a uniform way.

Reply:
sorry I pushed a change before I saw you replied. I'm fine with it staying a private method. we can just leave that as is. As for using `self` I'm surprised this works. I prefer to call static methods by their class to make the static nature a little more obvious and it also makes copy pasting easier for things like testing. (the self code shouldn't work in the REPL for example according to your SO question). Regardless I don't think this is a blocking issue so we can merge now as far as I'm concerned (though you'll actually have to approve it :P )

Reply:
LGTM